sansio_lsp_client/client.py

```diff
@@ -385,19 +388,22 @@ def parse_request(event_cls: t.Type[Event]) -> Event:
             progress_type = self._progress_tokens_map.get(request.params['token'])
 
             if progress_type == WorkDoneProgress:
-                kind = request.params.get('value', {}).get('kind')
-                if kind == 'begin':
+                kind = request.params['value']['kind']
+                kind = MWorkDoneProgressKind(kind)
+
+                if kind == MWorkDoneProgressKind.BEGIN:
                     return parse_request(WorkDoneProgressBegin)
-                elif kind == 'report':
+                elif kind == MWorkDoneProgressKind.REPORT:
                     return parse_request(WorkDoneProgressReport)
-                elif kind == 'end':
+                elif kind == MWorkDoneProgressKind.END:
                     del self._progress_tokens_map[request.params["token"]]
                     return parse_request(WorkDoneProgressEnd)
```

Comment:
Please add something like `else: raise RuntimeError(f"unknown kind {kind}")`

Reply:
``` python
kind = MWorkDoneProgressKind(kind) # line 392
```

^ this will raise an exception if kind is wrong:

`ValueError: 'hey' is not a valid MWorkDoneProgressKind`